backend/app/resources/auth_resources.py

```diff
@@ -30,9 +39,16 @@ def post(self):
             phone_number=data.get('phone_number', ''),
             password_hash=hashed_password_str
         )
+        new_user.id = id
         db.session.add(new_user)
         db.session.commit()
-        return USER_SCHEMA.dump(new_user), 201
+        user_schema = RegularUserSchema()
+        response = Response(
```

Comment:
Just one final comment, I don't think it's a good idea to instantiate a response object on every return statement, I think it's better to inject response object dependency via a constructor of a resource and re use it inside a resource  class. 

- [ ] handle response object dependency via constructor